plugins/modules/files/json_patch.py

```diff
@@ -0,0 +1,553 @@
+        chg = False
+        new_obj, removed, tst = self.remove(from_path, obj)
+        if removed is not None:  # don't inadvertently add 'None' as a value somewhere
+            new_obj, chg, tst = self.add(path, removed, new_obj)
+        return new_obj, chg, None
+
+    # https://tools.ietf.org/html/rfc6902#section-4.5
+    def copy(self, from_path, path, obj, **discard):
+        """Perform a 'copy' operation."""
+        value = self._get(from_path, obj)
+        if value is None:
+            raise PathError("could not find '%s' member in JSON object" % path)
+        new_obj, chg, tst = self.add(path, value, obj)
+        return new_obj, chg, None
+
+    # https://tools.ietf.org/html/rfc6902#section-4.6
+    def test(self, path, value, obj, **discard):
+        """Perform a 'test' operation.
+
+        This operation supports an additional feature not outlined in
+        RFC 6901 (https://tools.ietf.org/html/rfc6901): The ability to
+        reference every member of an array by using an asterisk (*).
+
+        In such a case, each member of the array at that point in the
+        path will be tested sequentially for the given value, and if
+        a matching value is found, the method will return immediately
+        with a True value.
+
+        Example:
+            {"op": "test", "path": "/array/*/member/property", "value": 2}
+
+        If the object to be tested looked like...
+        {
+            "array": [
+                {"member": {"property": 1}},
+                {"member": {"property": 2}}
+            ]
+        }
+        ... the result would be True, because an object exists within
+        "array" that has the matching path and value.
+        """
+        elements = path.lstrip('/').split('/')
+        next_obj = obj
+        for idx, elem in enumerate(elements):
+            if elem == "*":  # wildcard
+                if not isinstance(next_obj, list):
+                    return obj, None, False
+                for sub_obj in next_obj:
+                    dummy, chg, found = self.test('/'.join(elements[(idx + 1):]), value, sub_obj)
+                    if found:
+                        return obj, None, found
+                return obj, None, False
+            try:
+                next_obj = next_obj[elem]
+            except KeyError:
+                return obj, None, False
+            except TypeError:  # it's a list
+                if not elem.isdigit():
+                    return obj, None, False
+                try:
+                    next_obj = next_obj[int(elem)]
+                except IndexError:
+                    return obj, None, False
+        return obj, None, next_obj == value
+
+
+def main():
+    # Parsing argument file
+    module = basic.AnsibleModule(
+        argument_spec=dict(
+            src=dict(required=True, type='str'),
+            dest=dict(required=False, type='str'),
+            operations=dict(required=True, type='list', elements='?'),
```

Comment:
```suggestion
            operations=dict(required=True, type='list', elements='str'),
```
i put `?` because in this case it can be str or path, sorry for confusion

Reply:
Technically, I expect the following format for anything as an `operation` (I even have a validation method for it):
```yaml
operations:
  - op: add
    path: "/my/json/obj/attribute"
    value: "banana"
```
Reading through the docs, it does seem like I should be doing a second-level spec here?
```python
argument_spec=dict(
    operations=dict(required=True, type='list', elements=dict(
        op=dict(required=True, type='str', choices=['add', 'remove', 'replace', 'move', 'copy', 'test']),
        path=dict(required=True, type='str'),
        value=dict(required=False, type='raw'),
    )
)
```